src/internal/pjs/api_server.go

```diff
@@ -73,11 +74,20 @@ func (a *apiServer) CreateJob(ctx context.Context, request *pjs.CreateJobRequest
 
 func (a *apiServer) InspectJob(ctx context.Context, req *pjs.InspectJobRequest) (*pjs.InspectJobResponse, error) {
 	var jobInfo *pjs.JobInfo
+
+	if req.Job == nil {
```

Comment:
Ideally this would be a validation rule in the proto so that this code can avoid having to check.  I'm actually not sure that the GRPC machinery will call this with a nil message, but I haven't checked.  `proto.Equal((*pjs.Job)(nil), &pjs.Job{Id: 0})` should be true, btw, which is why this check might not matter.

Reply:
While gRPC _might_ not permit a nil `InspectJobRequest`, it will allow a non-nil message to contain a nil member (in this case, the `Job` member).

But lines 78–81 could be replaced with `id, err := a.resolveJob(ctx, req.Context, req.GetJob().GetId())`; `GetId` defaults to 0.